lib/Runtime/Library/JavascriptNumber.cpp

```diff
@@ -23,7 +23,7 @@ namespace Js
 #if FLOATVAR
         if (IsNan(value))
         {
-            value = JavascriptNumber::NaN;
+            value = IsNegative(value) ? JavascriptNumber::NegativeNaN : JavascriptNumber::NaN;
```

Comment:
>IsNegative [](start = 20, length = 10)

There might be other places where we canonicalize NaNs.
For example, `JavascriptConversion::TryCanonicalizeAsSimpleVar`.

Reply:
As well as JavascriptNumber::TryToVarFastWithCheck() and JavascriptNumber::NewWithCheck()

---
In reply to: [246601736](https://github.com/Microsoft/ChakraCore/pull/5905#discussion_r246601736) [](ancestors = 246601736)